pyfr/integrators/dual/pseudo/pseudosteppers.py

```diff
@@ -337,3 +337,309 @@ class DualRK45PseudoStepper(DualRKVdH2RPseudoStepper):
     @property
     def _pseudo_stepper_order(self):
         return 4
+
+
+class DualOptRKPseudoStepper(BaseDualPseudoStepper):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    @property
+    def _pseudo_stepper_has_lerrest(self):
+        return False
+
+    @property
+    def _stepper_nfevals(self):
+        return 7*self.nsteps
+
+    @property
+    def _pseudo_stepper_order(self):
+        return 1
+
+    @property
+    def _pseudo_stepper_nregs(self):
+        return 8
+
+    def step(self, t):
+        add = self._add
+        rhs = self._rhs_with_dts
+        dtau = self._dtau
+
+        # Get the bank indices for pseudo-registers (n+1,m; n+1,m+1; rhs),
+        # where m = pseudo-time and n = real-time
+        r0, r1, r2, r3, r4, r5, r6, r7 = self._pseudo_stepper_regidx
+
+        # Ensure r0 references the bank containing u(n+1,m)
+        if r0 != self._idxcurr:
+            r0, r1 = r1, r0
+
+        # First stage
+        rhs(t, r0, r1)
+
+        # Second stage
```

Comment:
A loop is probably best here — especially given the regularity in the second to eighth stage.